### DISCUSSION TITLE
x6100 remote control

### DISCUSSION THREAD

**Discussion by djbr1** [gdyuldin/x6100_gui#69] [Ideas]
Hi Zhenya,

Thanks for providing webserver for x6100.

Please consider developing "remote radio" capabilities:

- streaming audio input and output over wifi (internet).  Streaming output from radio was possible, dont know the status of sending audio input to x6100.  Wfview feature provided by xiegu a year ago was kind of step forward in that direction.

- remote display.  Qt (original implementation) allows VNC, either compiled or as startup parameter ( -platform vnc:port=5901:size=800x480 ).   I believe LVGL supports VNC.   

- "remote button press" .  Your LVGL implementation probably has API for command invocation, if not, it is possible to emulate physical button presses using "evemu", see my [gist](https://gist.github.com/djbr1/1ae3f822c139cd039b2cc559a29b1c21) made three years ago.  Although gist is cryptic (shows recording and replaying keypress of ATU button), I believe you get the idea.  Unlike wfview, "remote button press" may cover all available commands as on physical front panel

73
Branko
HB9TXB

PS. feel free to move this "issue" into Discussion/Ideas on https://github.com/gdyuldin/x6100_gui/

**Comment by gdyuldin** [comment]
Hi Branko.
I'm going to add support of wfview. I guess it will cover all requested features. 
I'm not sure about VNC. GUI was designed to this radio and will not be same handy with using on PC.
What buttons press you miss in the wfview?

73
Georgy // R2RFE

**Reply by qrzbing** [reply]
I am interested in wfview, is there any artical I can learn?
